### PR TITLE
Fixes doc and example

### LINKED ISSUE #412: Rough edges use case docs
I feel like a new user and in following the tutorial at https://eos.github.io/doc/use-cases.html#parameter-inference, I noticed a couple of rough edges that need some polishing. I will create a separate comment for each

### PR DESCRIPTION
**Pull request by mreboud**
Github: resolves #412

### CODE REVIEW

File: doc/use-cases.rst [hunk 1]
```diff
@@ -419,31 +420,31 @@ We then create an :class:`Analysis <eos.Analysis>` object as follows:
...
+        ],
+        'likelihood': [
+            'B->D::f_++f_0@HPQCD:2015A',
+            'B->D::f_++f_0@FNAL+MILC:2015B',
+            'B^0->D^+e^-nu::BRs@Belle:2015A',
+            'B^0->D^+mu^-nu::BRs@Belle:2015A'
+        ]
+    }
+    analysis = eos.Analysis(**analysis_args)
+    analysis.parameters['CKM::abs(V_cb)'].set(42.0e-3)
```
Comment (mreboud) [inline doc/use-cases.rst:441]: I need to rm whitespaces...

**Review by dvandyk — APPROVED** [review]
Looks good. We just discussed that adding a commit changing the comment on "run this in a notebook environment please" should also be added for the other two use cases.

File: doc/use-cases.rst [hunk 2]
```diff
@@ -419,31 +420,31 @@ We then create an :class:`Analysis <eos.Analysis>` object as follows:
...
-           'B^0->D^+e^-nu::BRs@Belle:2015A',
-           'B^0->D^+mu^-nu::BRs@Belle:2015A'
-       ]
-   }
-   analysis = eos.Analysis(**analysis_args)
-   analysis.parameters['CKM::abs(V_cb)'].set(42.0e-3)
+    analysis_args = {
+        'global_options': { 'form-factors': 'BSZ2015', 'model': 'CKM' },
+        'priors': [
+            { 'parameter': 'CKM::abs(V_cb)',           'min':  38e-3, 'max':  45e-3 , 'type': 'uniform'},
```
Comment (fredRos) [inline doc/use-cases.rst:426]: nitpick: 
```suggestion
            { 'parameter': 'CKM::abs(V_cb)',           'min':  38e-3, 'max':  45e-3, 'type': 'uniform'},
```
and similarlz below to save one character. Wohoo :wink:

File: doc/use-cases.rst [hunk 3]
```diff
@@ -526,6 +527,15 @@ equally across the full phase space. Instead, we equally distribute half of
 the points in the interval :math:`[0.02\,\text{GeV}^2, 1.00\,\text{GeV}^2]` and the other
 half in the remainder of the phase space.
 
+We produce :code:`N = 20000` samples with a thinning factor (or :code:`stride`) of 5. This means
```
Comment (fredRos) [inline doc/use-cases.rst:530]:
```suggestion
We run one Markov chain to produce :code:`N = 20000` samples with a thinning factor (or :code:`stride`) of 5. This means
```

File: eos/constraint.cc [hunk 4]
```diff
@@ -194,13 +194,12 @@ namespace eos
             Parameters parameters(Parameters::Defaults());
             ObservableCache cache(parameters);
 
-            for (const auto & po : this->options)
+            for (const auto & [key, value] : this->options)
             {
-                const auto & key = std::get<0>(po);
-                if (options.has(key))
+                if (options.has(key) && (value != options[key]))
```
Comment (fredRos) [inline eos/constraint.cc:199]: This hunk looks good

**Review by fredRos — APPROVED** [review]
Addresses part of my comments on #412 , Thanks for the fast fixes!

File: doc/use-cases.rst (hunk 1 above)
Comment (mreboud) [reply doc/use-cases.rst:441]: Hmm no in fact this is correct and was previously badly indented.

**Pull request merged**